app/controllers/suppliers_controller.rb

```diff
@@ -27,7 +27,7 @@ def new
 
   def create
     @supplier = Supplier.new(supplier_params)
-    @supplier.supplier_category ||= SupplierCategory.first
+    @supplier.supplier_category ||= SupplierCategory.first!
```

Comment:
This makes the server return a 404 for the whole page. I think this is not exactly what we aim for.

![Screenshot_2020-10-11 ActiveRecord RecordNotFound at f suppliers](https://user-images.githubusercontent.com/2647805/95677651-2b6ea100-0bc7-11eb-93f6-7cd6bfc66a33.png)

If we want a complete solution then we should create a default supplier category if none is found. Feels like overkill to me. What do you think?

Reply:
Hah, that makes sense. `supplier_category` [is not optional](https://github.com/foodcoops/foodsoft/blob/master/app/models/supplier.rb#L11). Since you can't delete SupplierCategories that have an associated Supplier, and the [initial migration](https://github.com/foodcoops/foodsoft/commit/d90d188dbf6b13bdfdbd92b34000b2c7e6a2b54c#diff-81ac1a9af21949a03f22bf0461a9a5a9) gives all Suppliers one, it is only possible for any last SupplierCategory to be deleted when there are no Suppliers. This is really an edge-case.

We could add a method `SupplierCategory.default` or so that returns the first or autocreated one, but I'm ok with leaving it like it is now, without the bang `!`.

Reply:
> This is really an edge-case.

Yes, I agree.

> We could add a method SupplierCategory.default or so that returns the first or autocreated one, [...]

I started to write `SupplierCategory.find_or_create_default!`, but a supplier category in turn [needs a financial transaction class](https://github.com/foodcoops/foodsoft/blob/3d5043fd052f79b33f8cae2ea3a5cc81ee972ba5/db/seeds/small.en.seeds.rb#L10). That approach will end up in several on-demand seeds in the models. Not sure if that is a good idea.

> [...] but I'm ok with leaving it like it is now, without the bang !.

Yep, I am convinced now that the bang is wrong there.